src/main/java/org/skife/jdbi/v2/Script.java

```diff
@@ -65,24 +62,55 @@
      */
     public void executeAsSeparateStatements() {
         for (String s : getStatements()) {
-            if (!WHITESPACE_ONLY.matcher(s).matches()) {
-                handle.execute(s);
-            }
+            handle.execute(s);
         }
     }
 
-    private String[] getStatements() {
+    private List<String> getStatements() {
         final String script;
         final StatementContext ctx = new ConcreteStatementContext(globalStatementAttributes);
-        try
-        {
+        try {
             script = locator.locate(name, ctx);
-        }
-        catch (Exception e)
-        {
+        } catch (Exception e) {
             throw new UnableToExecuteStatementException(String.format("Error while loading script [%s]", name), e, ctx);
         }
 
-        return script.replaceAll("\n", " ").replaceAll("\r", "").split(";");
+        return splitToStatements(script);
+    }
+
+    private List<String> splitToStatements(String script) {
+        List<String> statements = new ArrayList<String>();
+
+        StringBuilder sb = new StringBuilder();
+        SqlScriptLexer lexer = new SqlScriptLexer(new ANTLRStringStream(script));
+        boolean endOfFile = false;
+        while (!endOfFile) {
+            Token t = lexer.nextToken();
+            switch (t.getType()) {
+                case Token.EOF:
+                    addStatement(sb.toString(), statements);
+                    endOfFile = true;
+                    break;
+                case SqlScriptLexer.SEMICOLON:
+                    addStatement(sb.toString(), statements);
+                    sb = new StringBuilder();
+                    break;
+                default:
+                    sb.append(t.getText());
+                    break;
+                // We don't handle comments, because they are already removed in StatementLocator.
```

Comment:
I am worried that if this comment does not hold (say, someone changes `StatementLocator` and does not check this code) that non-SQL tokens may be appended to the buffer since they will hit the `default` case.  I would cover all cases explicitly here too and consider making the `default` case throw again.